dotnet/src/Connectors/Connectors.HuggingFace/Core/IStreamJsonParser.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
+
+/// <summary>
+/// Represents a JSON parser that can parse a Stream containing JSON data and yield the individual JSON objects.
+/// </summary>
+internal interface IStreamJsonParser
```

Comment:
I guess I should wait until it's no longer WIP, but I'm curious as to the purpose of all these newly defined internal interfaces

Reply:
This is an internal SSE streaming json parser for each object that contains a valid Json signature.